### DISCUSSION TITLE
Use the RenderAdminListData endpoint to enumerate sites?

### DISCUSSION THREAD

**Discussion by plamber** [pnp/pnpcore#1099] [Ideas]
Hi,
The new SharePoint Portal uses the /_api/SPO.Tenant/RenderAdminListData endpoint to retrieve sites in the admin portal.
Have you considered leveraging this endpoint instead of the current technique of reading the aggregated list?

Cheers,
Patrick

**Comment by jansenbe** [comment]
@plamber : interesting option, looking at the output it seems to be wrapper over the RenderListDataAsStream on top of the hidden lists in  Tenant Admin...I'll need to dig into the actual code to see why this was done and if there's a value add on using it in PnP. Thanks for pointing that out.

![image](https://user-images.githubusercontent.com/7451219/219337486-53a78719-a339-4300-88fc-628fccd1ff41.png)

**Reply by plamber** [reply]
I noticed that the updates performed on the list are slower than this endpoint's usage. For example, if you create a new site, when using our technique to enumerate sites it might take 10-15 minutes to return the proper results. The admin interface immediately shows the results without delay.
